File: clients/node/sdk/oauth/oauthRequestBuilder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OauthRequestBuilder = void 0;
const authorizeRequestBuilder_1 = require("./authorize/authorizeRequestBuilder");
const revokeRequestBuilder_1 = require("./revoke/revokeRequestBuilder");
const tokenRequestBuilder_1 = require("./token/tokenRequestBuilder");
const kiota_abstractions_1 = require("@microsoft/kiota-abstractions");
/** Builds and executes requests for operations under /oauth */
class OauthRequestBuilder {
    /** The authorize property */
    get authorize() {
        return new authorizeRequestBuilder_1.AuthorizeRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The revoke property */
    get revoke() {
        return new revokeRequestBuilder_1.RevokeRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The token property */
    get token() {
        return new tokenRequestBuilder_1.TokenRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /**
     * Instantiates a new OauthRequestBuilder and sets the default values.
     * @param pathParameters The raw url or the Url template parameters for the request.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    constructor(pathParameters, requestAdapter) {
        if (!pathParameters)
            throw new Error("pathParameters cannot be undefined");
        if (!requestAdapter)
            throw new Error("requestAdapter cannot be undefined");
        this.urlTemplate = "{+baseurl}/oauth";
        const urlTplParams = (0, kiota_abstractions_1.getPathParameters)(pathParameters);
        this.pathParameters = urlTplParams;
        this.requestAdapter = requestAdapter;
    }
    ;
}
exports.OauthRequestBuilder = OauthRequestBuilder;

File: clients/csharp/sdk/Oauth/OauthRequestBuilder.cs
using MastodonClientLib.Oauth.Authorize;
using MastodonClientLib.Oauth.Revoke;
using MastodonClientLib.Oauth.Token;
using Microsoft.Kiota.Abstractions;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using System;
namespace MastodonClientLib.Oauth {
    /// <summary>
    /// Builds and executes requests for operations under \oauth
    /// </summary>
    public class OauthRequestBuilder : BaseRequestBuilder {
        /// <summary>The authorize property</summary>
        public AuthorizeRequestBuilder Authorize { get =>
            new AuthorizeRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The revoke property</summary>
        public RevokeRequestBuilder Revoke { get =>
            new RevokeRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The token property</summary>
        public TokenRequestBuilder Token { get =>
            new TokenRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>
        /// Instantiates a new OauthRequestBuilder and sets the default values.
        /// </summary>
        /// <param name="pathParameters">Path parameters for the request</param>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        public OauthRequestBuilder(Dictionary<string, object> pathParameters, IRequestAdapter requestAdapter) : base(requestAdapter, "{+baseurl}/oauth", pathParameters) {
        }
        /// <summary>
        /// Instantiates a new OauthRequestBuilder and sets the default values.
        /// </summary>
        /// <param name="rawUrl">The raw URL to use for the request builder.</param>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        public OauthRequestBuilder(string rawUrl, IRequestAdapter requestAdapter) : base(requestAdapter, "{+baseurl}/oauth", rawUrl) {
        }
    }
}
